userland/userland_generic.ld

```diff
@@ -1,18 +1,21 @@
-ENTRY(_start)
-
-/* Userland Layout
- *
- * This script is designed to be used in conjunction with a chip_layout.ld from
- * a Tock board directory. A properly configured layout.ld file will look like:
- *
- *     INCLUDE <tock_board_dir>/chip_layout.ld
- *     INCLUDE <tock_userland_base_dir>./userland_layout.ld
+/* Userland Generic Layout
  *
- * The Tock build system should generate and use an appropriate linker script
- * automatically for you.
+ * This linker script is designed for Tock apps where the end microcontroller
+ * is not known. Therefore, this script over provisions space on some platforms.
  */
 
+/* Memory Spaces Definitions, 448K flash, 64K ram */
+ROM_ORIGIN  = 0x00010000;
+ROM_LENGTH  = 0x00020000;
+PROG_ORIGIN = 0x00030000;
+PROG_LENGTH = 0x00040000;
+RAM_ORIGIN  = 0x20000000;
+RAM_LENGTH  = 0x00010000;
 
+MPU_MIN_ALIGN = 8K;
```

Comment:
This is a problem for the NRF, where MIN_ALIGN is 8

More generally this is an example of a _chip-specific_ setting that's arch independent. The existence of these kinds of differences are a fundamental problem with the current approach. We either need to build a _per-chip_ image, or define some minimum specification that Tock-compatible chips must realize.